shared/lsif/definition.ts

```diff
@@ -0,0 +1,67 @@
+import * as sourcegraph from 'sourcegraph'
+import gql from 'tagged-template-noop'
+import { queryGraphQL as sgQueryGraphQL, QueryGraphQLFn } from '../util/graphql'
+import { LocationConnectionNode, nodeToLocation } from './locations'
+import {
+    GenericLSIFResponse,
+    queryLSIF,
+    resourceFragment,
+    rangeFragment,
+    lsifRequest,
+} from './api'
+
+export interface DefinitionResponse {
+    definitions: {
+        nodes: LocationConnectionNode[]
+    }
+}
+
+const definitionsQuery = gql`
```

Comment:
I don't know if there is a way to typecheck this, but it seems it would be really hard to debug if there is a typo or api change, what do you think?

Reply:
Added CI steps to ensure that the schema is up to date and the queries are validated against the schema.